### DISCUSSION TITLE
Usage in Python

### DISCUSSION THREAD

**Discussion by samjohnson1357** [LeelaChessZero/lc0#1799] [General]
Is there a way to use this chess engine in Python to simulate games between Leela Chess and perhaps some other engine (Stockfish)?  I've already tried using the so-much-meta/lczero_tools repo and it seems very outdated.

Thanks so much!

**Comment by mooskagh** [comment]
There is a `python-chess` (https://github.com/niklasf/python-chess) library that works well with all engines including Lc0.

You can find the example usage here: https://python-chess.readthedocs.io/en/latest/engine.html#analysing-and-evaluating-a-position.
You can also check how this library was used in our text user interface, here: https://github.com/LeelaChessZero/lc0-tui

**Reply by samjohnson1357** [reply]
Thanks so much!  This is exactly what I've been looking for.
